unitTests/ui/TemplateTest.ts

```diff
@@ -102,17 +104,44 @@ export function TemplateTest() {
         });
       });
 
-      it('should set the title attribute to the result title', async done => {
-        const created = await tmpl.instantiateToElement(result);
-        expect(created.title).toBe(result.title);
-        done();
-      });
+      describe(`when the template contains two ${ResultLink.ID} elements`, () => {
+        const resultLinkClass = Component.computeCssClassNameForType(ResultLink.ID);
+
+        async function initTemplateAndReturnResultLinks() {
+          const created = await tmpl.instantiateToElement(result);
+          return $$(created).findAll(`.${resultLinkClass}`);
+        }
+
+        beforeEach(() =>
+          (tmpl = new Template(() => {
+            return `
+          <div class="${resultLinkClass}"></div>
+          <div class="${resultLinkClass}"></div>
+          `;
+          })));
+
+        it('adds a role attribute with value "heading" to the first result link', async done => {
```

Comment:
Similar to my other comment, this should be tested inside the actual ResultLink component.